code/game/gamemodes/clock_cult/clock_cult.dm

```diff
@@ -115,43 +116,63 @@ Credit where due:
 	var/starter_servants = 3 //Guaranteed three servants
 	var/number_players = num_players()
 	roundstart_player_count = number_players
-	if(number_players > 30) //plus one servant for every additional 15 players
+	if(number_players > 30) //plus one servant for every additional 10 players above 30
 		number_players -= 30
-		starter_servants += round(number_players/15)
+		starter_servants += round(number_players/10)
+	starter_servants = min(starter_servants, 8) //max 8 servants (this should only happen with 110 players or more!)
 	while(starter_servants)
 		var/datum/mind/servant = pick(antag_candidates)
 		servants_to_serve += servant
 		antag_candidates -= servant
 		modePlayer += servant
+		servant.assigned_role = "Servant of Ratvar"
 		servant.special_role = "Servant of Ratvar"
-		servant.restricted_roles = restricted_jobs
 		starter_servants--
+	ark_time = 25 + round((roundstart_player_count / 5)) //In minutes, how long the Ark will wait before activation
+	ark_time = min(ark_time, 35) //35 minute maximum for the activation timer
 	return 1
 
 /datum/game_mode/clockwork_cult/post_setup()
 	for(var/S in servants_to_serve)
 		var/datum/mind/servant = S
 		log_game("[servant.key] was made an initial servant of Ratvar")
 		var/mob/living/L = servant.current
+		var/turf/T = pick(GLOB.servant_spawns)
+		L.forceMove(T)
+		GLOB.servant_spawns -= T
 		greet_servant(L)
 		equip_servant(L)
 		add_servant_of_ratvar(L, TRUE)
+	addtimer(CALLBACK(src, .proc/unlock_application_scripture), (ark_time * 0.5) * 600)
```

Comment:
just set it to a world.time and have the scripture_unlock_check() proc check if the world time is over it